CRM/Mailing/MailStore.php

```diff
@@ -57,12 +57,17 @@ public static function getStore($name = NULL) {
       throw new Exception("Empty mail protocol");
     }
 
+    // These changes make passwords with spaces, quotes, and backslashes work.
+    $password = str_replace('"', '\"', $dao->password);
+    $password = str_replace('\'', '\\', $password);
```

Comment:
it seems to me your replacing a \' with two \\ is that right?

Reply:
Well spotted seamus. After some extra testing involving combinations of special characters I was able to uncover a problem with that approach. I've now reverted back to addslashes with the comment @colemanw asked for. Happy to add unit tests as well if required.